lib/open_telemetry/tracer.ex

```diff
@@ -20,7 +20,8 @@ defmodule OpenTelemetry.Tracer do
   @type start_opts() :: %{
           optional(:parent) => OpenTelemetry.span() | OpenTelemetry.span_ctx(),
           optional(:attributes) => OpenTelemetry.attributes(),
-          optional(:sampler) => :ot_sampler.sampler(),
+          # TODO sampler should is an opaque type defined in the implementation
+          optional(:sampler) => term(),
```

Comment:
I'm not sure how to resolve this one. The type is defined here https://github.com/open-telemetry/opentelemetry-erlang/blob/master/src/ot_sampler.erl#L35 which is not a dependency of this library